documentation/docs/components/rest-controller.md

```diff
@@ -1 +1,59 @@
 # RESTController
+
+### Definition
+
+**_RESTController_** is used for making restful web services and handle the requests made by the client. It allows to handle all REST APIs such as GET, POST, DELETE, PUT, PATCH requests.
+**_RestController_** creates the request [DTO](https://bitloops.com/docs/bitloops-language/components/dto) and executes the [UseCase](https://bitloops.com/docs/bitloops-language/components/usecase) with this DTO created. Then, it just handles the result to present with the appropriate format and status code, the data to the client.
+
+### Examples
+
+##### Declaring a RESTController
+
+```typescript
+  RESTController UpdateTodoRESTController (updateTodoUseCase: UpdateTodoUseCase) {
+  method: REST.Methods.PUT;
+  execute(request, response) {
+    const dto = UpdateTodoRequestDTO({ id: request.params.id, completed: request.body.completed, title: request.body.title  });
+
+    const result = this.updateTodoUseCase.execute(dto);
+
+    if (result is Error) {
+      switch (result.getClass()) {
+        case ApplicationErrors.ToDoNotFoundError: {
+          this.notFound(response, result);
+          break;
+        }
+        default: {
+          this.fail(response, result.message);
+        }
+      }
+    } else {
+      this.ok(response);
+    }
+  }
+}
+```
+
+### Syntax
+
+##### Declaring a RESTController
+
+```typescript
+  RESTController <identifier name with a Controller suffix> (<dependencies>) {
+      method: REST.Methods.<REST API Methods>;
+      execute (<request>, <response>) {
+        <statements>
+        const result = this.useCase.execute(<DTO>);
```

Comment:
Here maybe we should have something more generic like `<useCase execution statement>` with reference to the useCase? So, that it does not seem to be mandatory to store it in result variable